docs/dev/documentation.md

```diff
@@ -1,5 +1,47 @@
 # Documentation
 
+This documentation is build with [Sphinx](https://www.sphinx-doc.org/) and
+hosted by [Read the Docs](https://about.readthedocs.com/)
+
+## Testing code snippets
+
+Almost all python code snippets in documentation tested using Sphinx's extension
+[sphinx.ext.doctest](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html).
+This allows to make sure that we have no typos, missed imports and other problems in:
+- code snippets marked with `testcode` directive in `*.md` files
+- code snippets from python's docstrings imported via `autoclass` directive in `*.rst` files
+
+CI pipeline is configured run Sphinx's `doctest` build automatically for each PR.
+It is also possible to run it locally:
+
+1. First you need to install docs requrements
+
+   ```bash
+   pip install -r requirements/docs.txt
+   ```
+
+2. Make sure you have `sphinx-build` command line tool avaliable in your venv.
```

Comment:
How is this not the case when installing the requirements first?

Reply:
Let's remove this step. I think it is result if my log investigation on what is `sphinx-build` tool and where it came from